cmd/kubeadm/app/cmd/upgrade/diff.go

```diff
@@ -121,7 +121,10 @@ func runDiff(flags *diffFlags, args []string) error {
 
 	cfg.ClusterConfiguration.KubernetesVersion = flags.newK8sVersionStr
 
-	specs := controlplane.GetStaticPodSpecs(&cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint)
+	specs, err := controlplane.GetAllStaticPodSpecs(&cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, true)
```

Comment:
this should be 
`specs, err := controlplane.GetAllStaticPodSpecs(&cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, true /* isDryRun */)`

Also we should add a comment here explaining why this is always set to dry-run true, because upgrde diff does not really change any state and only reports the diff between the existing static pod manifests and the static pod manifests that would be created on upgrade.